packages/framework/src/Model/Category/CategoryFactory.php

```diff
@@ -2,15 +2,27 @@
 
 namespace Shopsys\FrameworkBundle\Model\Category;
 
+use Shopsys\FrameworkBundle\Component\EntityExtension\EntityNameResolver;
+
 class CategoryFactory implements CategoryFactoryInterface
 {
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\EntityExtension\EntityNameResolver
+     */
+    private $entityNameResolver;
+
+    public function __construct(EntityNameResolver $entityNameResolver)
+    {
+        $this->entityNameResolver = $entityNameResolver;
+    }
 
     /**
      * @param \Shopsys\FrameworkBundle\Model\Category\CategoryData $data
      * @return \Shopsys\FrameworkBundle\Model\Category\Category
      */
     public function create(CategoryData $data): Category
     {
-        return new Category($data);
+        $categoryClassName = $this->entityNameResolver->resolve(Category::class);
+        return new $categoryClassName($data);
```

Comment:
We usually separate return statement with a blank line. In the past, this was (still is?) a part of the coding standards, albeit not automatically checked.

Reply:
Also, I would rather name the variable `$classData`, it would be easier for you to replicate in other entity factories.